content/posts/matplotlib/pyplot-vs-object-oriented-interface/index.md

```diff
@@ -88,7 +88,7 @@ plt.grid(True)
 
 The plot below shows us how the velocity is increasing.
 
-![png](figure/just-velocity.png)
+![Graph showing relationship between velocity and time for a free-falling object.](figure/just-velocity.png)
 
 <div class="image-caption">
 <b>Fig. 1.2</b> Velocity is increasing in fixed steps, due to a "constant" acceleration.
```

Comment:
@isabela-pf I took care of the merge conflict and merged @tupui other suggestion. But, I wasn't sure what to do about this. I am happy make the change, if you could let me know what you think about merging the caption and alt-text. More generally, is alt-text needed when there is a detailed caption? If so, how do you decide what goes in the caption and what does in the alt-text?

Thanks for all your and @MarsBarLee's help with this!

Reply:
Thanks for catching this @tupui, it's really important! 

## How it is now

I ran a screen reader (VoiceOver for me) over the above area's images (as they are now, with `png` as alt text) and captions and found that they are read as
- When focus is on image: "png, image"
- When focus is on caption: The caption is read like all other text, so in this case "Fig. 1.2…"

So in order, I get "png, image. Figure 1.2…" The captions are actually pretty good, so this isn't terrible but makes it unclear if the reader is missing information in this mystery image.

## What I suggested

I've added suggestions to label the images as their figure title so that it reads as "Fig. 1.2, image. Fig. 1.2 Velocity is increasing…" That may be redundant, but I think it makes it clear that the image is fully described by the following caption. 

If you have another idea, feel free to work with that; I'm happy to test it out. And thank you so so much for help with the merge conflict @jarrodmillman!